AssemblerLib/VectorMatrixAssembler.h

```diff
@@ -80,6 +80,46 @@ class VectorMatrixAssembler<GlobalMatrix, GlobalVector,
         local_assembler->addToGlobal(r_c_indices, M, K, b);
     }
 
+    /// Executes preTimestep call for each of the local assemblers for the
+    /// given mesh item.
+    /// The positions in the global matrix/vector are taken from
+    /// the LocalToGlobalIndexMap provided in the constructor at index \c id.
+    /// \attention The index \c id is not necesserily the mesh item's id.
+    template <typename LocalAssembler_>
+    void preTimestep(std::size_t const id,
+                     LocalAssembler_* const local_assembler,
+                     GlobalVector const& x) const
+    {
+        auto cb = [local_assembler](
+            std::vector<double> local_x,
+            LocalToGlobalIndexMap::RowColumnIndices const& r_c_indices)
+        {
+            local_assembler->preTimestep(local_x);
+        };
+
+        passLocalVector(cb, id, _data_pos, x);
```

Comment:
Forgot to pull your changes, but tested them in different branch. Stupid me.

I'll have to include the missing commit before other changes, so there will be a rebase, but the extra changes will be on the top.